src/Common/ZooKeeper/ZooKeeperImpl.cpp

```diff
@@ -446,8 +446,16 @@ void ZooKeeper::connect(
                 }
                 connected = true;
 
-                if (connected_callback.has_value())
-                    (*connected_callback)(i, node);
+                /// We can't user 'i' as idx because the nodes have already been shuffled.
+                for (size_t idx = 0; idx < args.hosts.size(); ++idx)
+                {
+                    /// The domain names of two nodes in a cluster cannot point to the same IP address.
```

Comment:
Well, two nodes can have the same IPs but different ports. However, `Poco::Net::SocketAddress` is host + port, so yes, it's unique